cpp/src/arrow/compute/kernels/scalar_compare_test.cc

```diff
@@ -1850,5 +1851,154 @@ TEST(TestMaxElementWiseMinElementWise, CommonTemporal) {
 }
 
+template <typename ArrowType>
+static void ValidateBetween(const Datum& val, const Datum& lhs, const Datum& rhs,
+                            const Datum& expected) {
+  ASSERT_OK_AND_ASSIGN(Datum result, Between(val, lhs, rhs));
+  AssertArraysEqual(*expected.make_array(), *result.make_array(),
+                    /*verbose=*/true);
+}
+
+template <typename ArrowType>
+static void ValidateBetween(const char* value_str, const Datum& lhs, const Datum& rhs,
+                            const char* expected_str) {
+  auto value = ArrayFromJSON(TypeTraits<ArrowType>::type_singleton(), value_str);
+  auto expected = ArrayFromJSON(TypeTraits<BooleanType>::type_singleton(), expected_str);
+  ValidateBetween<ArrowType>(value, lhs, rhs, expected);
+}
+
+template <>
+void ValidateBetween<StringType>(const char* value_str, const Datum& lhs,
+                                 const Datum& rhs, const char* expected_str) {
+  auto value = ArrayFromJSON(utf8(), value_str);
+  auto expected = ArrayFromJSON(TypeTraits<BooleanType>::type_singleton(), expected_str);
+  ValidateBetween<StringType>(value, lhs, rhs, expected);
+}
+
+template <typename ArrowType>
+class TestNumericBetweenKernel : public ::testing::Test {};
+
+TYPED_TEST_SUITE(TestNumericBetweenKernel, NumericArrowTypes);
+TYPED_TEST(TestNumericBetweenKernel, SimpleBetweenArrayScalarScalar) {
+  using ScalarType = typename TypeTraits<TypeParam>::ScalarType;
+  using CType = typename TypeTraits<TypeParam>::CType;
+
+  Datum zero(std::make_shared<ScalarType>(CType(0)));
+  Datum four(std::make_shared<ScalarType>(CType(4)));
+  ValidateBetween<TypeParam>("[]", zero, four, "[]");
+  ValidateBetween<TypeParam>("[null]", zero, four, "[null]");
+  ValidateBetween<TypeParam>("[0,0,1,1,2,2]", zero, four, "[0,0,1,1,1,1]");
+  ValidateBetween<TypeParam>("[0,1,2,3,4,5]", zero, four, "[0,1,1,1,0,0]");
+  ValidateBetween<TypeParam>("[5,4,3,2,1,0]", zero, four, "[0,0,1,1,1,0]");
+  ValidateBetween<TypeParam>("[null,0,1,1]", zero, four, "[null,0,1,1]");
+}
+
+TEST(TestSimpleBetweenKernel, SimpleStringTest) {
+  using ScalarType = typename TypeTraits<StringType>::ScalarType;
+  auto l = Datum(std::make_shared<ScalarType>("abc"));
+  auto r = Datum(std::make_shared<ScalarType>("zzz"));
+  ValidateBetween<StringType>("[]", l, r, "[]");
+  ValidateBetween<StringType>("[null]", l, r, "[null]");
+  ValidateBetween<StringType>(R"(["aaa", "aaaa", "ccc", "z"])", l, r,
+                              R"([false, false, true, true])");
+  ValidateBetween<StringType>(R"(["a", "aaaa", "c", "z"])", l, r,
+                              R"([false, false, true, true])");
+  ValidateBetween<StringType>(R"(["a", "aaaa", "fff", "zzzz"])", l, r,
+                              R"([false, false, true, false])");
+  ValidateBetween<StringType>(R"(["abd", null, null, "zzx"])", l, r,
+                              R"([true, null, null, true])");
+}
+
+TEST(TestSimpleBetweenKernel, SimpleTimestampTest) {
```

Comment:
Did you copy this from `TEST(TestCompareTimestamps, Basics)`?
Do you want to add between kernel version of `TEST(TestCompareTimestamps, DifferentParameters)` too?

Reply:
Will leave this out for now, Timestamps can be done in a follow up.

Reply:
Timestamp tests have been added.